src/vdom/diff.js

```diff
@@ -187,10 +187,10 @@ function innerDiffNode(dom, vchildren, context, mountAll, absorb) {
 		min = 0,
 		len = originalChildren.length,
 		childrenLen = 0,
-		vlen = vchildren && vchildren.length,
+		vlen = vchildren ? vchildren.length : 0,
 		j, c, vchild, child;
 
-	if (len) {
+	if (len!==0) {
```

Comment:
With the `?:` above the optimizing compiler knows that `len` is a number, so `len > 0` is fine. Shouldn't be any speed difference for this. Even the interpreted code is mostly equivalent from perf perspective.